--- /usr/lib/x86_64-linux-gnu/mutter-8/Clutter-8.gir	2021-08-09 09:59:52.000000000 -0600
+++ mutter-gir-files/Clutter-8.gir	2021-08-20 15:45:31.711844483 -0600
@@ -49161,36 +49161,6 @@
           </parameter>
         </parameters>
       </method>
-      <method name="add_primitive"
-              c:identifier="clutter_paint_node_add_primitive"
-              version="1.10"
-              introspectable="0">
-        <doc xml:space="preserve"
-             filename="../clutter/clutter/clutter-paint-node.c"
-             line="1034">Adds a region described by a Cogl primitive to the @node.
-
-This function acquires a reference on @primitive, so it is safe
-to call cogl_object_unref() when it returns.</doc>
-        <source-position filename="../clutter/clutter/clutter-paint-node.h"
-                         line="96"/>
-        <return-value transfer-ownership="none">
-          <type name="none" c:type="void"/>
-        </return-value>
-        <parameters>
-          <instance-parameter name="node" transfer-ownership="none">
-            <doc xml:space="preserve"
-                 filename="../clutter/clutter/clutter-paint-node.c"
-                 line="1036">a #ClutterPaintNode</doc>
-            <type name="PaintNode" c:type="ClutterPaintNode*"/>
-          </instance-parameter>
-          <parameter name="primitive" transfer-ownership="none">
-            <doc xml:space="preserve"
-                 filename="../clutter/clutter/clutter-paint-node.c"
-                 line="1037">a Cogl primitive</doc>
-            <type c:type="CoglPrimitive*"/>
-          </parameter>
-        </parameters>
-      </method>
       <method name="add_rectangle"
               c:identifier="clutter_paint_node_add_rectangle"
               version="1.10">
